### PR TITLE
加入模板帮助函数方式

### PR DESCRIPTION
**Pull request by wujingke**
加入系统模板帮助函数  和自定义注入全局函数  
注入方式 
app:conf('view func',{
    date = os.date,
    time =time2
}) 

修改一处os.time   为ngx.time

### CODE REVIEW

File: lib/lor/lib/utils/utils.lua [hunk 1]
```diff
@@ -44,4 +44,12 @@ function _M.random()
 end
 
 
-return _M
\ No newline at end of file
+function _M.table_merge( tDest, tSrc )
+    for k, v in pairs( tSrc ) do
+        tDest[k] = v
+    end
```
Comment (syhily) [inline lib/lor/lib/utils/utils.lua]: This would overwrite the same key in first table. any solution?

**Review by syhily — COMMENTED** [review]
A lot of unwanted format and blank line. `template_helper.substr` need more consideration

**Comment by syhily** [comment]
@sumory travis configutation need to be fixed.

File: lib/lor/lib/router/layer.lua [hunk 2]
```diff
@@ -5,7 +5,7 @@ local pairs = pairs
 local ipairs = ipairs
 local type = type
 local setmetatable = setmetatable
-local ostime = os.time
+local ostime = ngx.time
```
Comment (sumory) [inline lib/lor/lib/router/layer.lua]: os.time is only used for `math.randomseed` here. this small change will breaks most test cases for now.
maybe it's better to redefine `ostime` variable later.

Comment (syhily) [inline lib/lor/lib/router/layer.lua]: So we need a nginx stub for mock test like [this](https://github.com/tamqm/nginx-cdn/blob/master/Projects/bsm/fakengx/fakengx.lua)?

File: lib/lor/lib/view.lua [hunk 3]
```diff
@@ -39,15 +43,23 @@ function View:render(view_file, data)
         local view_file_name = view_file .. "." .. self.view_ext
 
         local t = template_new(view_file_name)
+
+        local view_func = utils.table_merge(template_helper,self.view_func)
```
Comment (sumory) [inline lib/lor/lib/view.lua]: other developers may have no idea about `template_helper` functions.

if they wanna use some `string` indeed while the inner `template_helper` take its place as a `function`,   they will be confused.

the users themselves should consider about the injected functions.

**Review by sumory — COMMENTED** [review]
please reformat some code style. 
some recommendations were made under the very code, please check.

**Comment by ms2008** [comment]
Closing this due to lack of activity. Please re-open if needed.

**Pull request closed without merging**